### PR TITLE
Flav/tables dsv wave 3

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

Extracted from https://github.com/dust-tt/dust/pull/7094.

The primary objective of the mentioned PR is to standardize the UI by using a single set of components, regardless of whether it's for a table or a document. Ultimately, the intent remains the same: allowing users to select a node in a tree. This PR enhances the existing tree components to manage both documents and tables. There are minor variations in how tree items are displayed depending on the view type. Additionally, I eliminated the extra calls to fetch parent nodes in the tree, as these are now included with the content nodes. This approach was necessary since the previous endpoint only worked with managed data sources and was malfunctioning anyway.

## Risk

Worst case it breaks the way we display the tree in both Assistant Builder and data source management.

### CODE REVIEW

File: front/components/assistant_builder/actions/ProcessAction.tsx [hunk 1]
```diff
@@ -301,22 +301,6 @@ export function ActionProcess({
     return null;
   }
 
-  const deleteDataSource = (name: string) => {
```
Comment (flvndvd) [inline front/components/assistant_builder/actions/ProcessAction.tsx:304]: This was dead code.

File: front/hooks/useParentResourcesById.ts [hunk 2]
```diff
@@ -1,78 +1,26 @@
-import type { DataSourceType, LightWorkspaceType } from "@dust-tt/types";
-import { useCallback, useEffect, useState } from "react";
-
-import type { GetContentNodeParentsResponseBody } from "@app/pages/api/w/[wId]/data_sources/[name]/managed/parents";
+import type { DataSourceViewContentNode } from "@dust-tt/types";
```
Comment (flvndvd) [inline front/hooks/useParentResourcesById.ts:1]: This is the most important change.

File: front/components/ConnectorPermissionsModal.tsx [hunk 3]
```diff
@@ -137,6 +137,7 @@ export function ConnectorPermissionsModal({
                   }));
                 }}
                 showExpand={CONNECTOR_CONFIGURATIONS[connector.type]?.isNested}
+                viewType="documents"
```
Comment (Fraggle) [inline front/components/ConnectorPermissionsModal.tsx:140]: worth a comment to explain why it's hardcoded ?

File: front/components/ConnectorPermissionsTree.tsx [hunk 4]
```diff
@@ -414,23 +423,27 @@ function PermissionTreeChildren({
   );
 }
 
+interface PermissionTreenProps {
```
Comment (Fraggle) [inline front/components/ConnectorPermissionsTree.tsx:426]: type: Treen vs Tree

File: front/components/assistant/AssistantDetails.tsx [hunk 5]
```diff
@@ -408,6 +408,7 @@ function DataSourceViewsSection({
                     setDocumentToDisplay(documentId);
                   }}
                   isSearchEnabled={false}
+                  viewType="documents"
```
Comment (Fraggle) [inline front/components/assistant/AssistantDetails.tsx:411]: same, hardcoded why ?

**Review by Fraggle — APPROVED** [review]
LGTM !

I think you can delete some table source selector related components, no ?

File: front/components/ConnectorPermissionsTree.tsx (hunk 4 above)
Comment (flvndvd) [reply front/components/ConnectorPermissionsTree.tsx:426]: Good catch!

**Pull request merged**